src/test/java/com/atlassian/bitbucket/jenkins/internal/client/BitbucketCapabilitiesClientImplTest.java

```diff
@@ -0,0 +1,39 @@
+package com.atlassian.bitbucket.jenkins.internal.client;
+
+import com.atlassian.bitbucket.jenkins.internal.client.supply.BitbucketCapabilitiesSupplier;
+import com.atlassian.bitbucket.jenkins.internal.model.AtlassianServerCapabilities;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class BitbucketCapabilitiesClientImplTest {
+
+    @Mock
+    private BitbucketCapabilitiesSupplier capabilitiesSupplier;
+    @Mock
+    private AtlassianServerCapabilities newCapabilities, cachedCapabilities;
+    @InjectMocks
+    private BitbucketCapabilitiesClientImpl capabilitiesClient;
+
+    @Test
+    public void testGetServerCapabilitiesNoCache() {
+        Mockito.when(capabilitiesSupplier.get()).thenReturn(newCapabilities);
+        assertEquals(newCapabilities, capabilitiesClient.getServerCapabilities());
+        Mockito.verify(capabilitiesSupplier).get();
+    }
+
+    @Test
+    public void testGetServerCapabilitiesWithCache() {
+        Mockito.when(capabilitiesSupplier.get()).thenReturn(cachedCapabilities);
+        capabilitiesClient.getServerCapabilities();
+        Mockito.verify(capabilitiesSupplier).get();
+
+        assertEquals(cachedCapabilities, capabilitiesClient.getServerCapabilities());
+        Mockito.verifyNoMoreInteractions(capabilitiesSupplier.get());
```

Comment:
I may be slightly misreading this, but I'm guessing this line was meant for verifying no more calls to `capabilitiesSupplier.get()`?
I _think_ what this verifies now (as opposed to `verifyNoMoreInteractions(capabilitiesSupplier)` for example), is that it says "verify no more interactions with the object/entity returned from `capabilitiesSupplier.get()`", i.e. no more interactions with `cachedCapabilities`.
There's no easy/graceful way to verify no more interactions with a specific method of a mock in Mockito that I know of unfortunately.

Good news is I just checked locally and `Mockito.verifyNoMoreInteractions(capabilitiesSupplier);` works in this case.

Reply:
Ah good catch! Thank you